web/src/components/troubleshoot/SupportBundleRow.tsx

```diff
@@ -1,44 +1,59 @@
 import * as React from "react";
-import { withRouter } from "react-router-dom";
+import { RouteComponentProps, withRouter } from "react-router-dom";
 import Loader from "../shared/Loader";
 import dayjs from "dayjs";
 import filter from "lodash/filter";
 import isEmpty from "lodash/isEmpty";
 import { Utilities } from "../../utilities/utilities";
 import download from "downloadjs";
 import Icon from "../Icon";
-// import { VendorUtilities } from "../../utilities/VendorUtilities";
-import { Repeater } from "../../utilities/repeater";
 import "@src/scss/components/AirgapUploadProgress.scss";
 
-let percentage;
+import {
+  KotsParams,
+  SupportBundle,
+  SupportBundleInsight,
+  SupportBundleProgress,
+} from "@types";
 
-class SupportBundleRow extends React.Component {
-  state = {
-    downloadingBundle: false,
-    downloadBundleErrMsg: "",
-    errorInsights: [],
-    warningInsights: [],
-    otherInsights: [],
-  };
+let percentage: number;
+
+type Props = {
+  bundle: SupportBundle;
+  isAirgap: boolean;
+  isCustomer: boolean;
+  isSupportBundleUploadSupported: boolean;
+  loadingBundle: boolean;
+  progressData: SupportBundleProgress;
+  refetchBundleList: () => void;
+  watchSlug: string;
+} & RouteComponentProps<KotsParams>;
+
+type State = {
+  downloadBundleErrMsg?: string;
+  downloadingBundle: boolean;
+  errorInsights?: SupportBundleInsight[];
+  otherInsights?: SupportBundleInsight[];
+  sendingBundle: boolean;
+  sendingBundleErrMsg?: string;
+  warningInsights?: SupportBundleInsight[];
+};
+
+class SupportBundleRow extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      downloadingBundle: false,
+      sendingBundle: false,
+    };
+  }
```

Comment:
added a constructor and removed all of the non boolean default values since they complicate the type definition.

Reply:
we don't need to set default values to non boolean states?